pages/blog/[article].js

```diff
@@ -0,0 +1,54 @@
+import fs from 'fs';
+import { join } from 'path';
+import dynamic from 'next/dynamic';
+import Content from 'components/Content/Content';
+import Head from 'components/head';
+import HeroBanner from 'components/HeroBanner/HeroBanner';
+import PropTypes from 'prop-types';
+
+const articlesDirectory = join(process.cwd(), 'blogArticles');
+
+function getAllArticleNames() {
+  const articleNames = fs.readdirSync(articlesDirectory);
+  return articleNames.map(name => {
+    return {
+      params: {
+        article: name.replace(/\.mdx$/, ''),
+      },
+    };
+  });
+}
+
+export async function getStaticPaths() {
+  return {
+    paths: getAllArticleNames(),
+    fallback: false,
+  };
+}
```

Comment:
Let's internalize he helper methods into `getStaticPaths` so they're _certainly_ only being defined/used in the server.